### PR TITLE
Raise Hyperspace Exception with a path information when JSON parse error has occurred

### PR DESCRIPTION
**Pull request by paryoja**
### What is the context for this pull request?

 - **Dependencies**: Links to issues you depend on for this pull request to work. Write N/A, if no dependencies.
   - Issue 1 : #373 
   - Issue 2 : #431

### What changes were proposed in this pull request?

Add error logs of the exact file that raises Exception to analyze the problem more easily.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Using UnitTest

### CODE REVIEW

**Comment by paryoja** [comment]
Since there are some issues regarding parsing Json logging files.
However, I found that it is not easy to pin point the problem since it is not easily reproducible.
So I added additional error log containing the log file path if there occurs exception while parsing the file.

@sezruby @imback82  I have some concerns within this PR.

1. Do you think this change is required? If so, am I following the rules regarding consensus of handling exceptions / logging errors?
2. Should I add unit test ensuring that the log is properly printed and containing the path when there are parsing errors?
3. My Scala formatter also changed lines in 57 / 105 with some indentations. Do you think my formatter setting is wrong? or it wasn't correct before?

**Comment by sezruby** [comment]
1 => good to show the exact file path, but it's better to throw the message with HyperspaceException as users need to check driver log to see the error message. There's no specific rules for logging/exception yet afaik.
2 => good to add a unit test, but it's not mandatory.  
3 => seems it wasn't correct before

**Comment by clee704** [comment]
@paryoja @sezruby Created #473 to handle the wrong formatting issue.

**Comment by paryoja** [comment]
> 1 => good to show the exact file path, but it's better to throw the message with HyperspaceException as users need to check driver log to see the error message. There's no specific rules for logging/exception yet afaik.

Ok. I will change it to HyperspaceException.
BTW, I feel like HyperspaceException covers quite broad area of Exceptions. Is there any plan for make subclasses for exceptions?

> 2 => good to add a unit test, but it's not mandatory.
I will make some unit test also.

File: src/main/scala/com/microsoft/hyperspace/index/IndexLogManager.scala [hunk 1]
```diff
@@ -71,8 +72,16 @@ class IndexLogManagerImpl(indexPath: Path, hadoopConfiguration: Configuration =
...
+
     val contents = FileUtils.readContents(fs, path)
-    Some(LogEntry.fromJson(contents))
+
+    try {
+      Some(LogEntry.fromJson(contents))
+    } catch {
+      case e: Exception =>
+        logError(s"Cannot parse json with file path: ${path}")
+        throw HyperspaceException(s"Cannot parse json with file path: ${path}")
```
Comment (sezruby) [inline src/main/scala/com/microsoft/hyperspace/index/IndexLogManager.scala:83]: Could you append e.getMessage() to the exception message?

File: src/main/scala/com/microsoft/hyperspace/index/IndexLogManager.scala [hunk 2]
```diff
@@ -71,8 +72,16 @@ class IndexLogManagerImpl(indexPath: Path, hadoopConfiguration: Configuration =
...
     }
+
     val contents = FileUtils.readContents(fs, path)
-    Some(LogEntry.fromJson(contents))
+
+    try {
+      Some(LogEntry.fromJson(contents))
+    } catch {
+      case e: Exception =>
+        logError(s"Cannot parse json with file path: ${path}")
```
Comment (sezruby) [inline src/main/scala/com/microsoft/hyperspace/index/IndexLogManager.scala:82]: This seems redundant as we throw the exception

File: src/main/scala/com/microsoft/hyperspace/index/IndexLogManager.scala [hunk 3]
```diff
@@ -71,8 +72,16 @@ class IndexLogManagerImpl(indexPath: Path, hadoopConfiguration: Configuration =
...
+
     val contents = FileUtils.readContents(fs, path)
-    Some(LogEntry.fromJson(contents))
+
+    try {
+      Some(LogEntry.fromJson(contents))
+    } catch {
+      case e: Exception =>
+        throw HyperspaceException(
+          s"Cannot parse json with file path: ${path}, reason: ${e.getMessage}")
```
Comment (sezruby) [inline src/main/scala/com/microsoft/hyperspace/index/IndexLogManager.scala:83]: No specific rule, but for consistency:
```suggestion
          s"Cannot parse JSON in ${path}: ${e.getMessage}")
```

Comment (paryoja) [reply src/main/scala/com/microsoft/hyperspace/index/IndexLogManager.scala]: Ok. Please let me know if the new title doesn't follow the consistency :)

**Review by sezruby — APPROVED** [review]
LGTM thanks!

**Pull request merged**